Source/settings/objects.js

```diff
@@ -739,10 +741,14 @@ FullScreenPokemon.FullScreenPokemon.settings.objects = {
                         { "thing": "InvisibleWall", "noBoundaryStretch": true }
                     ]
                 }
-            }
+            },
+            // State History
+            "state": {}
```

Comment:
Declaring `"state"` here means all instances of the object will have `.state` pointing to the same object. Such is how prototypal inheritance works.

Instead, have `addStateHistory` create `.state` if it doesn't exist.